### DISCUSSION TITLE
Target tracking using global_position instead of position

### DISCUSSION THREAD

**Discussion by m4rr5** [ramokz/phantom-camera#116] [Ideas]
In a project I am working on, I am integrating an external physics engine that uses a different axis orientation (Z+ up) from Godot itself (Y+ up). I deal with that by rotating my world in Godot by 90 degrees. This also means I need to treat the cameras a little differently but I've managed to make that all work except for one thing: target tracking.

After a bit of experimentation with the phantom-camera code it boils down to using get_global_position() instead of get_position() when retrieving the position of a tracked target. Of course this is a trivial change I could make myself (in a few places) but I am wondering if it would be possible to support my use case? For example by making this a parameter of PhantomCamera3D (use_global_position)?

**Comment by ramokz** [comment]
The usage of `get_position()` is actually a mistake that I just hadn't noticed... Should be fixed in the latest release where every instance is using `get_global_position()` instead.

Thanks for highlighting it!

**Reply by m4rr5** [reply]
You're welcome, thanks for sharing this great addon, it's extremely useful.

**Comment by m4rr5** [comment]
> The usage of `get_position()` is actually a mistake that I just hadn't noticed... Should be fixed in the latest release where every instance is using `get_global_position()` instead.
> 
> Thanks for highlighting it!

It looks like there are a few places you forgot in phantom_camera_3D.gd:
* https://github.com/ramokz/phantom-camera/blob/eac94c28a9870ef981d6417a63bc6fe5ac8e60a1/addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd#L558
* https://github.com/ramokz/phantom-camera/blob/eac94c28a9870ef981d6417a63bc6fe5ac8e60a1/addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd#L562
* https://github.com/ramokz/phantom-camera/blob/eac94c28a9870ef981d6417a63bc6fe5ac8e60a1/addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd#L564
* https://github.com/ramokz/phantom-camera/blob/eac94c28a9870ef981d6417a63bc6fe5ac8e60a1/addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd#L566

**Reply by ramokz** [reply]
Good spot! Have submitted another hotfix to address those.
That'll also teach me to not submit fixes during lunch breaks...

Thanks again!

**Reply by m4rr5** [reply]
You're welcome again! :)
